From 0e31533251937dac79465b68301a62125a03a4e0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jan-Simon=20M=C3=B6ller?= <dl9pf@gmx.de>
Date: Tue, 20 Nov 2012 21:19:50 +0100
Subject: [PATCH 1001/1005] cmpxchg-local-fix-uninitialized-return-code.patch

Fix uninitialized return code in default case in cmpxchg-local.h

This patch fixes warning when compiling the Linux kernel with Clang

Signed-off-by: Mark Charlebois <charlebm@gmail.com>
Signed-off-by: Behan Webster <behanw@converseincode.com>

---
 include/asm-generic/cmpxchg-local.h |    1 +
 1 file changed, 1 insertion(+)

--- linux.orig/include/asm-generic/cmpxchg-local.h
+++ linux/include/asm-generic/cmpxchg-local.h
@@ -41,6 +41,7 @@
 		break;
 	default:
 		wrong_size_cmpxchg(ptr);
+		uninitialized_var(prev);
 	}
 	raw_local_irq_restore(flags);
 	return prev;
